### ISSUE TITLE
AWS XRay compatibilities

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks-experiments/kspan#43]
I just thought it might be useful for folks to be know that the kspan trace IDs are currently not compatible with AWS XRay: https://github.com/open-telemetry/opentelemetry-collector-contrib/discussions/5257#discussioncomment-1343360

**Comment by bboreham** [comment]
Thanks for the report. 
Shouldn’t that be filed as an XRay incompatibility with OpenTelemetry?

**Comment by bboreham** [comment]
It is feasible to change the scheme kspan uses, at the cost of losing trace continuity across a restart of kspan. Or add some persistence mechanism to cover the restart.

**Comment by errordeveloper** [comment]
> Thanks for the report. 
> Shouldn’t that be filed as an XRay incompatibility with OpenTelemetry?

I updated the title.

**Comment by errordeveloper** [comment]
> It is feasible to change the scheme kspan uses, at the cost of losing trace continuity across a restart of kspan. Or add some persistence mechanism to cover the restart. 

Yes, it's neat to have restart persistence by the virtue of consistent hashing. Maybe someone could think of this as a separate mode. By the way, the timestamps seem to be mandated by the W3C spec also...
